internal/orchestrator/orchestrator.go

```diff
@@ -359,14 +364,24 @@ func (or *orchestrator) initPlugins(ctx context.Context) (err error) {
 		return err
 	}
 
-	if or.publicstorage == nil {
-		psType := config.GetString(config.PublicStorageType)
-		if or.publicstorage, err = psfactory.GetPlugin(ctx, psType); err != nil {
-			return err
+	if or.sharedstorage == nil {
+		ssType := config.GetString(config.SharedStorageType)
+		if or.sharedstorage, err = ssfactory.GetPlugin(ctx, ssType); err != nil {
+			if strings.Contains(err.Error(), i18n.Expand(ctx, i18n.MsgUnknownSharedStoragePlugin, "")) {
+				// backward compatibility for old "publicstorage" name
+				ssType = config.GetString(config.PublicStorageType)
+				if or.sharedstorage, err = ssfactory.GetPlugin(ctx, ssType); err != nil {
+					return err
+				}
+			} else {
+				return err
+			}
 		}
 	}
-	if err = or.publicstorage.Init(ctx, publicstorageConfig.SubPrefix(or.publicstorage.Name()), or); err != nil {
-		return err
+	if err = or.sharedstorage.Init(ctx, sharedstorageConfig.SubPrefix(or.sharedstorage.Name()), or); err != nil {
+		if err = or.sharedstorage.Init(ctx, publicstorageConfig.SubPrefix(or.sharedstorage.Name()), or); err != nil {
```

Comment:
I think you should also track which config was used above and invoke the matching prefix, rather than simply trying both.